test/test_utils/ec2.py

```diff
@@ -904,3 +909,43 @@ def post_process_mxnet_ec2_performance(connection, log_location):
         return {"Throughput": total / n}
     else:
         raise ValueError("total: {}; n: {} -- something went wrong".format(total, n))
+
+
+@retry(reraise=True, wait=wait_fixed(60), stop=stop_after_attempt(5))
+def install_python_in_instance(context, python_version="3.9"):
```

Comment:
Could we add a precheck to see if the python version we want is already available? As written it looks like we install blindly.

Reply:
Makes sense, but I think it might be better to do a blind install of pyenv always, because `dist-packages` always make things more difficult for us.

Reply:
Works for me. I suppose the additional overhead is pretty minimal.